src/scripts/core/models/timer.ts

```diff
@@ -0,0 +1,52 @@
+import { Container, Text as PixiText } from "pixi.js";
+import { gsap } from "gsap";
+import { eventEmitter } from "../../../event-emitter";
+
+export class Timer extends Container{
+
+    private _totalTime: number;
+    private _currentTime: number;
+    private _timeText: PixiText = new PixiText;
+
+    public get totalTime() {
+        return this._totalTime;
+    }
+
+    constructor(totalTime: number) {
+        super()
+        this._totalTime = totalTime;
+        this._currentTime = totalTime;
+        this.formatTime();
+        this._timeText.style = {
+            fill: "white"
+        };
+        this.addChild(this._timeText);
+        this.countdown();
+    }
+
+    private formatTime(): void {
+        let minutes = "0";
+        let seconds = "0";
+        if(this._currentTime > 60){
+            minutes = Math.floor(this._currentTime/60).toString();
+        }
+        seconds = this._currentTime % 60 === 0 ? '00' : (this._currentTime % 60).toString()
```

Comment:
Is this showing seconds less than 10 as '1' or '01'?

I could be missing something but I'd expect this to be showing as '1' but you have the check to convert '0' to '00' so reads like it should be '01'.

[You may want to look into the padStart function to better handle this.](https://www.w3schools.com/jsref/jsref_string_padstart.asp)